From ea34d282dbf23298d6f007c52ffdfb3b434d4908 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Mon, 14 Oct 2024 17:16:35 +0100
Subject: [PATCH 5/6] c-family: Advertise the availability attribute support in
 features.

This is a temporary workaround - we are going to advertise the availablility
attribute as present for all targets - which it is not.  For the Darwin
branch it is a short-term band-aid.

This addresses Issue #12 part 2.

Signed-off-by: Iain Sandoe <iains@gcc.gnu.org>

(cherry picked from commit 843342fa4391c163bb6522839317d81f01c035de)
---
 gcc/c-family/c-common.cc | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git gcc/c-family/c-common.cc gcc/c-family/c-common.cc
index e3c3fae8dea7..c4ee5c88b9f4 100644
--- gcc/c-family/c-common.cc
+++ gcc/c-family/c-common.cc
@@ -346,7 +346,11 @@ static constexpr hf_feature_info has_feature_table[] =
   { "enumerator_attributes",		  HF_FLAG_NONE, 0 },
   { "tls",				  HF_FLAG_NONE, 0 },
   { "gnu_asm_goto_with_outputs",	  HF_FLAG_EXT, 0 },
-  { "gnu_asm_goto_with_outputs_full",	  HF_FLAG_EXT, 0 }
+  { "gnu_asm_goto_with_outputs_full",	  HF_FLAG_EXT, 0 },
+  { "attribute_availability",		  HF_FLAG_NONE, 0 },
+  { "attribute_availability_with_message",  HF_FLAG_NONE, 0 },
+  { "attribute_availability_with_replacement",  HF_FLAG_NONE, 0 },
+  { "attribute_availability_with_version_underscores",  HF_FLAG_NONE, 0 }
 };
 
 /* Global visibility options.  */
-- 
2.47.0

